app/models/course_option.rb

```diff
@@ -78,6 +78,29 @@ def part_time?
     study_mode == 'part_time'
   end
 
+  def self.find_through_api(course_data)
```

Comment:
```suggestion
  def self.find_through_api!(course_data)
```
Given this method calls a bang method, we should show that in the method name. This will highlight that this method will raise an error

Reply:
Apologies for leading you down this path - it looks like the Offer endpoint use a service called `GetCourseOptionFromCodes` to do the same look up - we should lean on that for this lookup functionality 🙃